linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker
+
+import scala.concurrent._
+
+import java.nio.charset.StandardCharsets.UTF_8
+
+import org.junit.Test
+import org.junit.Assert._
+
+import org.scalajs.ir.ClassKind
+import org.scalajs.ir.Names._
+import org.scalajs.ir.Trees._
+import org.scalajs.ir.Types._
+
+import org.scalajs.logging._
+
+import org.scalajs.junit.async._
+
+import org.scalajs.linker._
+import org.scalajs.linker.interface._
+
+import org.scalajs.linker.testutils._
+import org.scalajs.linker.testutils.TestIRBuilder._
+
+class IncrementalTest {
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  import IncrementalTest._
+
+  @Test // #4335
+  def testChangeMethodReachedFromExport(): AsyncResult = await {
+
+    val FooClass = ClassName("Foo")
+
+    val jsMethodName = StringLiteral("foo")
+    val staticMethodName = m("value", Nil, IntRef)
+
+    def classDefs(pre: Boolean) = Seq(
+        mainTestClassDef(
+            consoleLog(JSMethodApply(LoadModule(FooClass), jsMethodName, Nil))
+        ),
+        classDef(
+            FooClass,
+            kind = ClassKind.ModuleClass,
+            superClass = Some(ObjectClass),
+            memberDefs = List(
+                trivialCtor(FooClass),
+                JSMethodDef(EMF, jsMethodName, Nil,
+                    if (pre) int(5)
+                    else ApplyStatic(EAF, FooClass, staticMethodName, Nil)(IntType)
+                )(EOH, None),
```

Comment:
Subtle formatting rule: if the `)` is on a new line, then there must be a newline after the `(`:
```suggestion
                JSMethodDef(
                    EMF, jsMethodName, Nil,
                    if (pre) int(5)
                    else ApplyStatic(EAF, FooClass, staticMethodName, Nil)(IntType)
                )(EOH, None),
```
That said, usually we format non-Block IR nodes using running arguments, including for the additional parameter lists:
```suggestion
                JSMethodDef(EMF, jsMethodName, Nil,
                    if (pre) int(5)
                    else ApplyStatic(EAF, FooClass, staticMethodName, Nil)(IntType))(
                    EOH, None),
```